[48기 배정완] Add: 댓글 기능 구현 완료

wan0514 opened the pull request:
## 1. 본 PR이 우리 팀의 웹 서비스 제품성에 어떠한 기여를 하였고, 사용자에게 어떠한 기대효과를 전달하는지 작성해주세요.
- 내 PR이 제품 내 어떠한 기능적인 배경/전후맥락 가운데 개발되었나요?

  (예시) 고객의 유입과 전환을 책임지는 회원 기능에서, 유저의 이탈을 방지하기 위해 input 창 바로 아래에서 실시간으로 적색의 경고 메세지를 전달 받습니다.

- 내 PR이 Merge 됨으로써 유저에게 전달되는 편익/기대효과는 무엇일까요?

  (예시) 유저는 실시간으로 스스로 오기입한 내용을 확인할 수 있기 때문에 잘못 계정 정보를 작성함으로써 발생될 수 있는 변수를 최소화할 수 있습니다. 이는 곧 유저 전환의 허들을 낮추어 줄 수 있습니다.



<br />

## 2. 이 브랜치에서 어떤 내용을 개발했는지 큰 제목과 그리고 상세 내역을 적어주세요.

(예시) 로그인 기능 추가

- (예시) email입력 `@`필요,password에 숫자로 8자이상 유효성 검사 추가
- (예시) 유효성 검사 통과시 버튼 컬러 변경 및 버튼 활성화

<br />

## 3. 개발한 화면을 캡쳐해서 첨부 해 주세요. (drag & drop 또는 첨부파일 추가)
<br />

-

<br />

## 4. 이 브랜치에서 개발하면서 느꼇던 개발 성장포인트를 적어주세요.

- (예시) 새로운 배열메서드 filter를 쓸 수 있게됐습니다.
- (예시) 상수데이터와 map 메서드로 반복되는 UI를 그릴 수 있게 됐습니다.
- (예시) fetch함수를 어떻게 쓰는지 알게됐습니다.
- (예시) 계산된 속성명 (computed property)을 쓰긴했는데 어떻게 돌아가는지 이해가 잘 안되서
  블로그에 정리해봐야겠습니다.
- (예시) 화살표 함수(arrow-function) 어떤 코드를 언제 생략할 수있는지 완벽 이해했습니다.
- (예시) 구조분해할당(Destructuring assignment)을 써보니 가독성도, 유지보수도 편할것같습니다.
  <br />

wanwork committed:
Add: 기존 작업 move

wanwork committed:
Modify: index.html 팀원들과 싱크 맞추기

wanwork committed:
Modify: favicon.ico 교체

wanwork committed:
Add : Mission 1 - 사용자 입력 데이터 저장 기능 구현

wanwork committed:
Add : Mission 2 - 로그인 버튼 활성화 기능 구현

wanwork committed:
"ADD: 댓글 작성 레이아웃 구현"
- 기존 댓글작성 영역을  <div> 에서  <textarea>로 변경
- margin  수정

wanwork committed:
"Add : 댓글 작성시 게시 버튼 노출 구현"

wanwork committed:
Modify : 코드 리팩토링

5yoonl commented on src/pages/jeongwan/Login/Login.js:
```diff
@@ -0,0 +1,75 @@
+
+  //사용자 입력 데이터 저장
+
+  const [id, setId] = useState("");
+  const [pw, setPw] = useState("");
```
Props & State 리뷰세션에서 진행한 
스프레드 연산자와 계산된속성명을 활용해서 state를 하나로 줄이는 과정과 handling하는 함수도 하나로 줄여보는 연습을 해보셨으면 좋겠습니다.
스프레드 연산자와 계산된속성명은 앞으로 개발을 하실때 굉장히 자주 사용하게 될 내용이라서 당장 이해가 되지 않더라도 눈과 손에 익히는 과정이 굉장히 중요합니다

5yoonl commented on src/pages/jeongwan/Login/Login.js:
```diff
@@ -0,0 +1,75 @@
+  const [pw, setPw] = useState("");
+
+  const saveUserId = (event) => {
+    // console.log(event.target.value);
+    setId(event.target.value);
+    // console.log(event.target.value);
```
test가 끝난 console은 주석으로 남겨놓지않고 꼭 삭제해주세요!

5yoonl commented on src/pages/jeongwan/Login/Login.js:
```diff
@@ -0,0 +1,75 @@
+
+  return (
+    <div className="login">
+      <div id="content_wrap">
+        <h1 className="logo_name">westagram</h1>
```
className을 작성할 땐 camelCase로 작성해주세요!
```suggestion
      <div id="contentWrap">
        <h1 className="logoName">westagram</h1>
```

5yoonl commented on src/pages/jeongwan/Main/Main.js:
```diff
@@ -0,0 +1,378 @@
+import { useState } from "react";
+
+const Main = () => {
+  // const menu_element = [{
+  //   id : 1,
+  //   class : menu_element
+  //   img : ""
+  //   title : 홈
+  // },{}]
+
+  // menu_element.map(item,index){
+  //   console.log(item);
+
+  //   <div>
+  //     <a>
+  //       <div></div>
+  //       <span></span>
+  //     </a>
+  //   </div>
+  // };
+
+  // 댓글 수에 따라 높이 자동 조절
+
+  // const textarea = useRef;
+
+  // const handleResizeHeight = () => {
+  //   textarea.current.style.height = "auto";
+  //   textarea.current.style.height = textarea.current.scrollHeight + "px";
+  // };
+
+  //실패
+
+  // Input이 생기면 게시 버튼 생기기 없으면 다시 사라지기
```
불필요하거나 코드를 설명하기위한 주석이 많습니다.
어려운 코드를 주석으로 설명하기 보다 처음부터 이해하기 쉬운코드를 작성할 수 있도록 노력해주세요!
추가적인 개인학습을 위해 남겨놓은 주석이라면 개인 메모장이나 블로그를 활용해주세요!

5yoonl commented on src/pages/jeongwan/Main/Main.js:
```diff
@@ -0,0 +1,378 @@
+                    className="comment_input text_s text_gray"
+                    placeholder="댓글 달기..."
+                  ></textarea>{" "}
+                  {isCommentInput ? (
+                    <div onClick="addComment" className="inputButton text_bold">
+                      게시
+                    </div>
+                  ) : null}
```
조건부 랜더링!! 👍 좋습니다!

5yoonl reviewed (commented):
과제진행하느라 수고많으셨습니다 정완님!
코드에 주석을 많이 남겨놓는것도 나중엔 나쁜습관으로 자리잡을 수 있습니다.
지금부터 사용하지 않는 코드라면 과감히 지우거나 꼭 필요한 내용이라면 개인적인 공간에 옮겨적어, github에는 올라오지 않을 수 있도록 관리해주세요!

wanwork committed:
Add : Mission 3 - 댓글 기능 구현

wanwork committed:
Add : Mission 4 - 댓글 컴포넌트화, Props로 데이터 전달 구현

wanwork committed:
“Modify: 컴포넌트 폴더 생성 및 분리 완료, Aside에서 상수데이터 활용 완료”
- Main 폴더 하위에 Components 폴더 생성후 컴포넌트 파일 이동
- Aside에서 상수데이터 활용 완료. 개별 파일로 분리 미적용 상태

wanwork committed:
Modify: 상수 데이터를 별도의 파일로 분리하여 동일하게 구현

wanwork committed:
Modify: Feed 컴포넌트 생성

wanwork committed:
Modify:className CamelCase로 수정 완료

wanwork committed:
Modify: Nav 상수데이터터 활용
